### PR TITLE
Updated DQMOffline/Muon package for Run3 

### PR DESCRIPTION
**Pull request by calderona**
Some minor changes have been implemented in view of the next Run3 data taking and certification:

- Removed the DTSegmentMonitor folder (currently not used by the muon POG) 
- Changes in PU ranges to extend them for next Run3 and modified for Phase2
- Minor changes for some plot names 

It has been tested on the following samples: 

34850 ZMM, No PU, 14TeV, 2026 upgrade 
1330.0 ZMM, noPU, 13TeV standard
136.867 RunSingleMu2018B, standard

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1
Changes expected and validated by the developer

File: DQMOffline/Muon/python/EfficencyPlotter_cfi.py [hunk 1]
```diff
@@ -100,10 +100,55 @@
...
                                           )
 
 
+effPlotter=cms.Sequence(effPlotterLoose*effPlotterMedium*effPlotterTight)
+effPlotter_miniAOD=cms.Sequence(effPlotterLooseMiniAOD*effPlotterMediumMiniAOD*effPlotterTightMiniAOD)   
+
+effPlotterLoose_Phase2=effPlotterLoose.clone()
+effPlotterLoose_Phase2.vtxBin=20
+effPlotterLoose_Phase2.vtxMin=149.5
+effPlotterLoose_Phase2.vtxMax=249.5
```
Comment (perrotta) [inline DQMOffline/Muon/python/EfficencyPlotter_cfi.py:116]:
```suggestion
effPlotterLoose_Phase2=effPlotterLoose.clone(
  vtxBin=20
  vtxMin=149.5
  vtxMax=249.5
)
```
And the same for the other cases below

File: DQMOffline/Muon/python/EfficencyPlotter_cfi.py [hunk 2]
```diff
@@ -100,10 +100,55 @@
...
+
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/EfficencyPlotter_cfi.py:118]: You have only one empty line in the other cases: remove this one

File: DQMOffline/Muon/python/EfficencyPlotter_cfi.py [hunk 3]
```diff
@@ -100,10 +100,55 @@
...
+effPlotterMedium_Phase2=effPlotterMedium.clone()                                                                                            
+effPlotterMedium_Phase2.vtxBin=20                                                                                                           
+effPlotterMedium_Phase2.vtxMin=149.5                                                                                                        
+effPlotterMedium_Phase2.vtxMax=249.5 
+
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/EfficencyPlotter_cfi.py:124]: You have only one empty line in the other cases: remove this one

File: DQMOffline/Muon/python/EfficencyPlotter_cfi.py [hunk 4]
```diff
@@ -100,10 +100,55 @@
...
+effPlotterTightMiniAOD_Phase2.vtxMin=149.5                                                                                                  
+effPlotterTightMiniAOD_Phase2.vtxMax=249.5        
+
+effPlotter_Phase2=cms.Sequence(effPlotterLoose_Phase2*effPlotterMedium_Phase2*effPlotterTight_Phase2)
+effPlotter_miniAOD_Phase2=cms.Sequence(effPlotterLooseMiniAOD_Phase2*effPlotterMediumMiniAOD_Phase2*effPlotterTightMiniAOD_Phase2)                                    
+                                                                                                             
+from Configuration.Eras.Modifier_phase2_muon_cff import phase2_muon                                                                         
+phase2_muon.toReplaceWith(effPlotter,effPlotter_Phase2)     
+phase2_muon.toReplaceWith(effPlotter_miniAOD,effPlotter_miniAOD_Phase2)  
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/EfficencyPlotter_cfi.py:151]: Empty lines at the end are not needed

File: DQMOffline/Muon/python/muonAnalyzer_cff.py [hunk 5]
```diff
@@ -45,7 +37,9 @@
                                   staMuonSegmentAnalyzer*
                                   muonKinVsEtaAnalyzer*
                                   diMuonHistos*
-                                  LooseMuonEfficiencyAnalyzer*
-                                  MediumMuonEfficiencyAnalyzer*
-                                  TightMuonEfficiencyAnalyzer*                                
+                                  EfficiencyAnalyzer* 
                                   muonPFsequence)
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonAnalyzer_cff.py:42]: Empty lines at the end are not needed

File: DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py [hunk 6]
```diff
@@ -170,11 +171,61 @@
...
+
+
+EfficiencyAnalyzer = cms.Sequence(TightMuonEfficiencyAnalyzer*LooseMuonEfficiencyAnalyzer*MediumMuonEfficiencyAnalyzer)
+
+EfficiencyAnalyzer_miniAOD = cms.Sequence(TightMuonEfficiencyAnalyzer_miniAOD*LooseMuonEfficiencyAnalyzer_miniAOD*MediumMuonEfficiencyAnalyzer_miniAOD)
+
+TightMuonEfficiencyAnalyzer_Phase2=TightMuonEfficiencyAnalyzer.clone()
+TightMuonEfficiencyAnalyzer_Phase2.vtxBin=20
+TightMuonEfficiencyAnalyzer_Phase2.vtxMax=249.5
+TightMuonEfficiencyAnalyzer_Phase2.vtxMin=149.5
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py:189]:
```suggestion
TightMuonEfficiencyAnalyzer_Phase2=TightMuonEfficiencyAnalyzer.clone(
  vtxBin=20
  vtxMax=249.5
  vtxMin=149.5
  )
```
and the same here below

File: DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py [hunk 7]
```diff
@@ -170,11 +171,61 @@
...
+LooseMuonEfficiencyAnalyzer_Phase2=LooseMuonEfficiencyAnalyzer.clone()                                                                      
+LooseMuonEfficiencyAnalyzer_Phase2.vtxBin=20                                                                                                
+LooseMuonEfficiencyAnalyzer_Phase2.vtxMin=149.5  
+LooseMuonEfficiencyAnalyzer_Phase2.vtxMax=249.5
+
+MediumMuonEfficiencyAnalyzer_Phase2=MediumMuonEfficiencyAnalyzer.clone()                                                                    
+MediumMuonEfficiencyAnalyzer_Phase2.vtxBin=20                                                                                               
+MediumMuonEfficiencyAnalyzer_Phase2.vtxMin=149.5  
+MediumMuonEfficiencyAnalyzer_Phase2.vtxMax=249.5 
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py:200]: Remove extra empty line

File: DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py [hunk 8]
```diff
@@ -170,11 +171,61 @@
...
+
+TightMuonEfficiencyAnalyzer_miniAOD_Phase2=TightMuonEfficiencyAnalyzer_miniAOD.clone()                                                      
+TightMuonEfficiencyAnalyzer_miniAOD_Phase2.vtxBin=20                                                                                       
+TightMuonEfficiencyAnalyzer_miniAOD_Phase2.vtxMax=249.5                                                                                     
+TightMuonEfficiencyAnalyzer_miniAOD_Phase2.vtxMin=149.5                                                                                    
+     
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py:207]: Remove extra empty line

File: DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py [hunk 9]
```diff
@@ -170,11 +171,61 @@
...
+EfficiencyAnalyzer_Phase2 = cms.Sequence(TightMuonEfficiencyAnalyzer_Phase2*LooseMuonEfficiencyAnalyzer_Phase2*MediumMuonEfficiencyAnalyzer_Phase2)
+
+EfficiencyAnalyzer_miniAOD_Phase2 = cms.Sequence(TightMuonEfficiencyAnalyzer_miniAOD_Phase2*LooseMuonEfficiencyAnalyzer_miniAOD_Phase2*MediumMuonEfficiencyAnalyzer_miniAOD_Phase2)                                                                                                    
+
+            
+from Configuration.Eras.Modifier_phase2_muon_cff import phase2_muon                                                                         
+phase2_muon.toReplaceWith(EfficiencyAnalyzer, EfficiencyAnalyzer_Phase2)                                                                   
+phase2_muon.toReplaceWith(EfficiencyAnalyzer_miniAOD, EfficiencyAnalyzer_miniAOD_Phase2)    
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py:227]: Remove extra empty lines

File: DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py [hunk 10]
```diff
@@ -170,11 +171,61 @@
...
+MediumMuonEfficiencyAnalyzer_miniAOD_Phase2.vtxBin=20                                                                                       
+MediumMuonEfficiencyAnalyzer_miniAOD_Phase2.vtxMin=149.5                                                                                    
+MediumMuonEfficiencyAnalyzer_miniAOD_Phase2.vtxMax=249.5                                                                                   
+                                                        
+
+EfficiencyAnalyzer_Phase2 = cms.Sequence(TightMuonEfficiencyAnalyzer_Phase2*LooseMuonEfficiencyAnalyzer_Phase2*MediumMuonEfficiencyAnalyzer_Phase2)
+
+EfficiencyAnalyzer_miniAOD_Phase2 = cms.Sequence(TightMuonEfficiencyAnalyzer_miniAOD_Phase2*LooseMuonEfficiencyAnalyzer_miniAOD_Phase2*MediumMuonEfficiencyAnalyzer_miniAOD_Phase2)                                                                                                    
+
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonEfficiencyAnalyzer_cfi.py:223]: Remove extra empty line

File: DQMOffline/Muon/python/muonIsolationDQM_cff.py [hunk 11]
```diff
@@ -43,11 +49,31 @@
...
-                              directory = cms.string("Muons/Isolation/global")
+                              directory = cms.string("Muons/Isolation/global"),
+                             vtxBin = cms.int32(30),
+                             vtxMax = cms.double(149.5),
+                             vtxMin = cms.double(0.5)
                               )
 muIsoDQM_seq = cms.Sequence(MuIsoDQM_trk+MuIsoDQM_sta+MuIsoDQM_glb)
 
 
+MuIsoDQM_glb_Phase2=MuIsoDQM_glb.clone()
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonIsolationDQM_cff.py:60]: As above...

File: DQMOffline/Muon/python/muonIsolationDQM_cff.py [hunk 12]
```diff
@@ -88,6 +120,33 @@
...
-                                      directory = cms.string("Muons_miniAOD/Isolation/global")
+                                      directory = cms.string("Muons_miniAOD/Isolation/global"),
+                                     vtxBin = cms.int32(30),
+                                     vtxMax = cms.double(149.5),
+                                     vtxMin = cms.double(0.5),
                                       )
 muIsoDQM_seq_miniAOD = cms.Sequence(MuIsoDQM_trk_miniAOD+MuIsoDQM_sta_miniAOD+MuIsoDQM_glb_miniAOD)
+
+
+MuIsoDQM_glb_miniAOD_Phase2=MuIsoDQM_glb_miniAOD.clone()
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonIsolationDQM_cff.py:131]: As above

File: DQMOffline/Muon/python/muonMonitors_cff.py [hunk 13]
```diff
@@ -35,3 +34,5 @@
 
 
 muonMonitorsAndQualityTests = cms.Sequence(muonMonitors*muonQualityTests)
+
```
Comment (perrotta) [inline DQMOffline/Muon/python/muonMonitors_cff.py:37]: Remove extra empty lines

**Comment by calderona** [comment]
Updated all the requested modifications in the code

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please abort

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**